src/neural-network.js

```diff
@@ -874,25 +758,12 @@ export default class NeuralNetwork {
       const node = layer[nodeKey];
       let result = [node.bias];
       for (let w in node.weights) {
-        if (node.weights[w] < 0) {
-          result.push(`${node.weights[w]}*(${nodeHandle(layers, layerNumber - 1, w)})`);
-        } else {
-          result.push(`+${node.weights[w]}*(${nodeHandle(layers, layerNumber - 1, w)})`);
-        }
+        result.push(
+          `${node.weights[w] < 0 ? '' : '+'}${node.weights[w]}*(${nodeHandle(layers, layerNumber - 1, w)})`
+        );
       }
 
-      switch (activation) {
-        case 'sigmoid':
-          return `1/(1+1/Math.exp(${result.join('')}))`;
-        case 'relu':
-          return `var sum = ${result.join('')};(sum < 0 ? 0 : sum);`;
-        case 'leaky-relu':
-          return `var sum = ${result.join('')};(sum < 0 ? 0 : 0.01 * sum);`;
-        case 'tanh':
-          return `Math.tanh(${result.join('')});`;
-        default:
-          throw new Error('unknown activation type ' + activation);
-      }
+      return `(${availableActivations[activation].activationFn.toString()})(${result.join('')})`;
```

Comment:
@robertleeplummerjr Thank you. 🙏 

The one thing that i came my mind is to replace `sum`s with `result...` on activation function body and return it as a string. But this would also make code hard to maintain since it's hard to predict what would be added as new activation function in the future. Could be really complex activation that would cause changes on design of method - availableActivations - itself. Using same functions here would save us from unpredicted bugs and hardcoding complex activations IMO.

AFAIK, IIFE is good friend of memory since they immediately being released by mark-and-sweep garbage collectors on JavaScript engine. It might not be as bad as you think but i can't argue with it unless i have some benchmark data. :) 

So, i'll give it a try and compare performances of functions generated by current version and my version. I'll let you know about results.

Do you have any other suggestion here?